pkg/storageincentives/agent.go

```diff
@@ -321,7 +326,21 @@ func (a *Agent) play(ctx context.Context) (uint8, []byte, error) {
 	}
 
 	t := time.Now()
-	sample, err := a.sampler.ReserveSample(ctx, salt, storageRadius, uint64(time.Now().UTC().UnixNano()))
+	block, err := a.backend.BlockNumber(ctx)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	previousRoundNumber := (block / a.blocksPerRound) - 1
+
+	timeLimiterBlock, err := a.backend.HeaderByNumber(ctx, new(big.Int).SetUint64(previousRoundNumber*a.blocksPerRound))
+	if err != nil {
+		return 0, nil, err
+	}
+
+	timeLimiterNanoseconds := timeLimiterBlock.Time * 1000000
```

Comment:
assuming timeLimiterBlock.Time is in seconds, the following is more clear
`time.Second * (timeLimiterBlock.Time) / time.Nanosecond`